design/cluster-status-conditions.md

```diff
@@ -0,0 +1,39 @@
+# Cluster Status Condition
+
+## Background
+
+Currently, Ceph Cluster stores the state of the system in `Status.State`.However OLM expects the state to be in `Status.Phase`. The elimination of State variable breaks the backward compatibility of the Ceph Cluster. Although keeping both of them is also not a good practice. But the usage of `Status.Phase` is deprecated over time because it contradicts the system design principle and hampered evolution. So rather than encouraging clients to infer the implicit properties from phases, the usage of `Status.Condition` is preferred.
```

Comment:
Let's keep this generic and remove OLM from the argument. Changes proposed by this document would anyway not meet OLM's expectation.

We could say something like,
Status.State is being used. But, we want to move to Status.Conditions. We do not want to move to Status.Phase because it is now deprecated and even kubernetes apis are shifting to Status.Conditions.

Probably add some more reasoning and comparison so that others can benefit.

Reply:
Fixed this.